packages/logger/src/logger/Logger.ts

```diff
@@ -58,15 +58,16 @@ function isLoggerFlushed(logger: AugmentedLogger): boolean {
 // This async function can be called by a bot if the log message is generated right before the process terminates.
 // This method will check if all transports attached to AugmentedLogger having isFlushed getter return it as true. If
 // not, it will block until such time that all these transports have been flushed. This still can exit before all
-// transports are flushed if the logger flush timeout is reached.
+// transports are flushed if the logger flush timeout is reached for non-persistent log queue transports.
 export async function waitForLogger(logger: AugmentedLogger): Promise<void> {
-  pauseLogQueueProcessing(logger.transports); // Signal to pause log queue processing.
-
   const waitForFlushed = async (): Promise<void> => {
     while (!isLoggerFlushed(logger)) await delay(0.5); // While the logger is not flushed, wait for it to be flushed.
```

Comment:
Sorry, I know this is not related to this PR:
Is there a reason for using a tight loop with a constant 0.5-second delay in waitForLogger? This could lead to high CPU usage without a back-off mechanism. Could we consider implementing an increasing delay to mitigate this?

`while (!isLoggerFlushed(logger)) await delay(Math.min(currentDelay += 0.5, maxDelay))`

Reply:
Agree that this could lead to higher CPU usage, but a better solution would be for any flushable transport to emit event once it can be terminated and just await for that event here. I could create an abstract FlushableTransport class for this in a follow-up